src/lib/services/collectionService.ts

```diff
@@ -0,0 +1,334 @@
+import { useQuery } from "@tanstack/react-query";
+import axios from "axios";
+
+import { CELATONE_QUERY_KEYS, useCelatoneApp } from "lib/app-provider";
+import {
+  getCollectioUniqueHoldersCount,
+  getCollectionActivitiesCount,
+  getCollectionActivitiesPagination,
+  getCollectionByCollectionAddress,
+  getCollectionCreator,
+  getCollectionListByAddress,
+  getCollectionMutateEventsCount,
+  getCollectionMutateEventsPagination,
+  getCollectionTotalBurnedCount,
+  getCollectionsPagination,
+} from "lib/query";
+import type { HexAddr } from "lib/types";
+
+import type {
+  ActivitiesCountResponse,
+  CollectionCreator,
+  CollectionCreatorResponse,
+  CollectionInfo,
+  CollectionInfoResponse,
+  MutationEventsCountResponse,
+  CollectionResponse,
+  TotalBurnedCountResponse,
+  UniqueHoldersCountResponse,
+  ActivitiesResponse,
+  Activity,
+  Collection,
+  CollectionMutateEventsResponse,
+  CollectionMutateEvent,
+  CollectionListByAddressResponse,
+  CollectionByAddress,
+} from "./collection";
+import {
+  zActivitiesCountResponseItem,
+  zCollectionCreatorResponseItem,
+  zCollectionInfoResponseItem,
+  zCollectionResponseItem,
+  zTotalBurnedResponseItem,
+  zUniqueHoldersCountResponseItem,
+  zActivitiesResponseItem,
+  zMutationEventsCountResponseItem,
+  zCollectionMutateEventsResponse,
+  zCollectionListByAddressResponse,
+} from "./collection";
+
+export const useCollectionsPagination = (
+  pageSize: number,
+  offset: number,
+  search?: string
+) => {
+  const { chainConfig } = useCelatoneApp();
+
+  const queryFn = async () => {
+    return axios
+      .post<CollectionResponse>(chainConfig.indexer, {
+        query: getCollectionsPagination,
+        variables: { offset, pageSize, search },
+      })
+      .then(({ data: res }) =>
+        res.data.collections.map((collection) =>
+          zCollectionResponseItem.parse(collection)
+        )
+      );
+  };
```

Comment:
Let's develop a query function within collection.ts. This way, we can avoid importing the zod schema and response type directly into this file."